src/convert_json.cpp

```diff
@@ -43,17 +43,21 @@ Json::Value fetchJsonValue(const std::string &url,
 	httpfetch_sync(fetch_request, fetch_result);
 
 	if (!fetch_result.succeeded) {
-		return Json::Value();
+		std::string err_msg = "HTTP request failed with code ";
+		err_msg += itos(fetch_result.response_code);
+		throw JsonFetchException(err_msg.c_str());
```

Comment:
Maybe JsonFetchException could have a http_response_code member added to it.  Perhaps there could even be a JsonFetchException::getFriendlyErrorString().